terraform-provider-constellation/internal/provider/attestation_data_source.go

```diff
@@ -87,6 +102,33 @@ func (d *AttestationDataSource) Schema(_ context.Context, _ datasource.SchemaReq
 	}
 }
 
+// ValidateConfig validates the configuration for the image data source.
+func (d *AttestationDataSource) ValidateConfig(ctx context.Context, req datasource.ValidateConfigRequest, resp *datasource.ValidateConfigResponse) {
+	var data AttestationDataSourceModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if !data.AttestationVariant.Equal(types.StringValue("azure-sev-snp")) && !data.MaaURL.IsNull() {
+		resp.Diagnostics.AddAttributeError(
+			path.Root("maa_url"),
+			"MAA URL can only be set for Azure SEV-SNP", "When using other attestation variants, it should not be set.",
+		)
+		return
+	}
+
+	if data.AttestationVariant.Equal(types.StringValue("azure-sev-snp")) && data.MaaURL.IsNull() {
+		resp.Diagnostics.AddAttributeError(
+			path.Root("maa_url"),
+			"MAA URL must be set for Azure SEV-SNP", "Only when attestation_variant is set to 'azure-sev-snp', 'maa_url' must be specified.",
```

Comment:
```suggestion
	if data.AttestationVariant.Equal(types.StringValue("azure-sev-snp")) && data.MaaURL.IsNull() {
		resp.Diagnostics.AddAttributeError(
			path.Root("maa_url"),
			"MAA URL should be set for Azure SEV-SNP", "Only when attestation_variant is set to 'azure-sev-snp', 'maa_url' must be specified.",
```
afaik MAA URL is not a hard requirement, and Constellation works just fine without it.
Using the CLI, one can also configure Azure Constellation to not use it at all (which might be desired as using the MAA requires placing some trust into Azure's services).